subprojects/core/src/integTest/groovy/org/gradle/api/LazyAdditionToDomainObjectCollectionIntegrationTest.groovy

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api
+
+import groovy.test.NotYetImplemented
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class LazyAdditionToDomainObjectCollectionIntegrationTest extends AbstractIntegrationSpec {
+    def setup() {
+        buildFile << """
+            abstract class Base {}
+            abstract class Sub extends Base {}
+            
+            def container = objects.domainObjectSet(Base) 
+            
+            def create(type) {
+                def ret = objects.newInstance(type)
+                println "created " + ret
+                return ret
+            }
+        """
+    }
+
+    def "addLater(Base) with eager all"() {
```

Comment:
The names on many of these track the test logic, but don't really express the purpose of the test very clearly.

Maybe soemthing like: 
"allLater() triggers logic configured in .all()"
"allLater() triggers logic configured in .withType() on base container type"
"allLater() on sub type does NOT trigger logic configured in .withType() on base container type", etc.

Reply:
Thanks, I renamed all of these methods